src/oemof/solph/_options.py

```diff
@@ -160,6 +167,18 @@ def _check_age_and_lifetime(self):
                 )
                 raise AttributeError(e4)
 
+    def _check_nonconvex(self):
+        """Checking for unnecessary setting of nonconvex"""
+        if self.nonconvex:
+            if (self.minimum.min() == 0) and (self.offset.min() == 0):
+                msg = (
+                    "It is not necessary to set the investment to `nonconvex` "
+                    "if `minimum` and `offset` are 0.\n"
+                    "This can lead to the `invest_status` variable becoming "
+                    "1, even if the `nominal_capacity` is optimized to 0."
+                )
+                warn(msg, debugging.SuspiciousUsageWarning)
+
```

Comment:
You should should check if the `SuspiciousUsageWarning` is triggered. The according test could go to `tests/test_options.py`.